Source/Core/Section.swift

```diff
@@ -232,8 +232,8 @@ extension Section : MutableCollection, BidirectionalCollection {
         }
     }
 
-    public subscript (range: Range<Int>) -> [BaseRow] {
-        get { return kvoWrapper.rows.objects(at: IndexSet(integersIn: range)) as! [BaseRow] }
+    public subscript (range: Range<Int>) -> ArraySlice<BaseRow> {
+        get { return ArraySlice<BaseRow>(kvoWrapper.rows.objects(at: IndexSet(integersIn: range)) as! [BaseRow]) }
```

Comment:
@mcousillas6 @m-revetria  I think ArraySlice should keep index from original collection, in this case the range. This array slice start and end index are 0 . number of items in the collection instead of range.startIndex, range.EndIndex.